### PR TITLE
Feature/oi stats v2

### PR DESCRIPTION
**Pull request by Steve-Rog**
This PR is about graph ingrafting for displaying IO data. Also added filters to display data in the last month/week/year.

Screenshot 2023-09-18 alle 09 32 23

I used Dune to do the data fetch, I had to use a specially created API key, so it will definitely need to be changed and moved (maybe into an .env)

### CODE REVIEW

File: v2/perps-v2/ui/src/hooks/useOiStats.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+        const lastRow = transformedRowsFormatted[transformedRowsFormatted.length - 1];
+        const totalLongShortAbsoluteValue = {
+          longTotal: lastRow.long,
+          shortTotal: lastRow.short,
+          absoluteValue: lastRow.long - lastRow.short,
+        };
+
+        setTotalShortLoss(totalLongShortAbsoluteValue);
+        setLastRow(lastRow);
```
Comment (bachstatter) [inline v2/perps-v2/ui/src/hooks/useOiStats.ts:107]: I dont think this LastRow state is used / needed

File: v2/perps-v2/ui/src/hooks/useOiStats.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  day: string;
+}
+
+interface ShortLoss {
+  shortTotal: number;
+  longTotal: number;
+  absoluteValue: number;
+}
+
+const UseOiStats = (DUNE_API_KEY: string, period: 'W' | 'M' | 'Y') => {
```
Comment (bachstatter) [inline v2/perps-v2/ui/src/hooks/useOiStats.ts:22]: lowercase this one

File: v2/perps-v2/ui/src/utils/constants.ts [hunk 3]
```diff
@@ -7,6 +7,8 @@ export const KWENTA_SUBGRAPH_URL = 'https://api.thegraph.com/subgraphs/name/kwen
 export const POLYNOMIAL_SUBGRAPH_URL =
   'https://api.thegraph.com/subgraphs/name/synthetix-perps/polynomial';
 
+export const DUNE_API_KEY = 'IRdMBz8n8CzLbjH33EHuRHe0lJifJab9';
```
Comment (bachstatter) [inline v2/perps-v2/ui/src/utils/constants.ts:10]: Yea talking to a fellow CC about some solution for this. Note that this key is now exposed and anyone can you it to make request to dune.

File: v2/perps-v2/ui/src/hooks/useOiStats.ts (hunk 1 above)
Comment (Steve-Rog) [reply v2/perps-v2/ui/src/hooks/useOiStats.ts]: committed changes

File: v2/perps-v2/ui/src/hooks/useOiStats.ts (hunk 2 above)
Comment (Steve-Rog) [reply v2/perps-v2/ui/src/hooks/useOiStats.ts]: committed changes

Comment (Steve-Rog) [reply v2/perps-v2/ui/src/hooks/useOiStats.ts]: I used fetch as indicated [here](https://github.com/Synthetixio/js-monorepo/pull/959) instead of axios

Comment (Steve-Rog) [reply v2/perps-v2/ui/src/hooks/useOiStats.ts]: Added hook for date formatting as [here](https://github.com/Synthetixio/js-monorepo/pull/959)

File: v2/perps-v2/ui/src/utils/constants.ts (hunk 3 above)
Comment (jmzwar) [reply v2/perps-v2/ui/src/utils/constants.ts:10]: Yeah we'll need to remove the key @bachstatter let's catch up for a solution to this as we don't have OI historical data in the sg

**Comment by jmzwar** [comment]
@Steve-Rog closing this

**Pull request closed without merging**